src/vscode-dts/vscode.proposed.toolProgress.d.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	export interface ProgressOptions2 extends Omit<ProgressOptions, 'location'> {
+		location: ProgressLocation | {
+			/**
+			 * The identifier of a view for which progress should be shown.
+			 */
+			viewId: string;
+		} | {
+			/**
+			 * An invocation token for progress shown while a {@link LanguageModelTool} is running.
+			 */
+			toolInvocationToken: ChatParticipantToolToken | undefined;
```

Comment:
This is creative 😄 I think I might prefer something that just works through the tool invocation itself though? It emits a stream or calls a progress.report or something like that. Curious to hear from Joh or others but it just feels a bit off somehow that I would pass the special token into this other progress API.

Reply:
Agreeing with Rob. Apart from coupling tool invocations with progress locations this seems overly complex. 

Didn't think much about streams but IMO this should be  `LanguageModelTool#invoke(options, token, progress)` or `LanguageModelToolInvocationOptions#progress`.

That also guarantees only legit tool invocations can report progress but the API is better contained, easier to discover and IMO way simpler to implement. Let's start over on this